core/src/apps/cardano/sign_tx.py

```diff
@@ -213,11 +211,14 @@ def _validate_outputs(
     total_amount = 0
     for output in outputs:
         total_amount += output.amount
-        if output.address_parameters:
-            # try to derive the address to validate it
-            derive_address_bytes(
-                keychain, output.address_parameters, protocol_magic, network_id
+
+        if output.address_parameters and output.address is not None:
```

Comment:
Perhaps compare `output.address_parameters` with `None` explicitly?

Reply:
I don't think this is needed as this is the way objects are checked throughout the project and also in the next condition. Address is checked against `None` since it's a string. I would keep it as is.